core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2171,21 +2189,41 @@ private[spark] class DAGScheduler(
                         abortStage(mapStage, reason, None)
                       } else {
                         rollingBackStages += mapStage
+                        mapOutputTracker.unregisterAllMapAndMergeOutput(
+                          mapStage.shuffleDep.shuffleId)
                       }
+                    } else {
+                      mapOutputTracker.unregisterAllMapAndMergeOutput(
+                        mapStage.shuffleDep.shuffleId)
```

Comment:
Add if/else for `SHUFFLE_USE_OLD_FETCH_PROTOCOL`.
Orthogonally, fiven we are in 4.0, perhaps it is time to drop `SHUFFLE_USE_OLD_FETCH_PROTOCOL` @attilapiros  ?